CONTRIBUTING_JS.md

```diff
@@ -100,6 +101,25 @@ However, we've added an extra linting rule: Enforce the use of [strict mode](htt
 
 Using [aegir-lint](#aegir) will help you do this easily; it automatically lints your code.
 
+#### Error Codes
+
+When introducing a new error code that may be useful outside of the current scope, make sure it is defined in a named variable and can be exported and used in other places. This enables others to reuse those definitions and decreases the number of hardcoded values across our codebases. For example:
+
+```js
+const {  Errors } = require('interface-datastore')
+
+// throw predefined errors if possible
+if (!value) {
+  throw Errors.notFoundError()
+}
+
+// compare value from 
+const ERR_NOT_FOUND = Errors.notFoundError().code 
+if (err.code === ERR_NOT_FOUND) {
+  // handle
+}
+```
+
```

Comment:
I've seen this style in `js-ipfs-repo` which I quite liked and means you don't have to create an error to get the code and exclude the error creation function from the stack:

```js
class LockExistsError extends Error {
  constructor (message) {
    super(message || 'Lock already exists')
    this.name = 'LockExistsError'
    this.code = LockExistsError.code
  }
}

LockExistsError.code = 'ERR_LOCK_EXISTS'
exports.LockExistsError = LockExistsError
```

Then usage:

```js
const { LockExistsError } = require('ipfs-repo')

// throw predefined errors if possible
if (!value) {
  throw new LockExistsError()
}

// compare value from 
if (err.code === LockExistsError.code) {
  // handle
}
```

What do you think?

Reply:
👍 I also prefer exposing the code statically.

Reply:
Yes, this is better :+1: 
I updated example in https://github.com/ipfs/community/pull/445/commits/29f16bb476c1a4d727b40036243fc5dcf8550083